packages/clay-color-picker/src/Custom.tsx

```diff
@@ -0,0 +1,240 @@
+}
+
+/**
+ * Renders the custom color picker
+ */
+const Custom: React.FunctionComponent<CustomProps> = ({
+	colors,
+	label,
+	onChange,
+	onColorsChange,
+	spritemap,
+	value,
+}) => {
+	const color = tinycolor(value);
+
+	const [activeSplotchIndex, setActiveSplotchIndex] = useState(0);
+	const [hue, setHue] = useState(0);
+	const [editorActive, setEditorActive] = useState(false);
+	const [hexInputVal, setHexInput] = useHexInput(color.toHex());
+
+	const {b, g, r} = color.toRgb();
+	const {s, v} = color.toHsv();
+
+	const rgbArr: [number, string][] = [[r, 'r'], [g, 'g'], [b, 'b']];
+
+	const setNewColor = (colorValue: tinycolor.Instance, setInput = true) => {
+		const hexString = colorValue.toHex();
+
+		const newColors = [...colors];
+
+		newColors[activeSplotchIndex] = hexString;
+
+		onColorsChange(newColors);
+
+		onChange(hexString);
+
+		if (setInput) {
+			setHexInput(colorValue.toHex());
+		}
+	};
+
+	return (
+		<React.Fragment>
+			{label && (
+				<div className="clay-color-header">
+					<span className="component-title">{label}</span>
+
+					<button
+						className={`${
+							editorActive ? 'close' : ''
+						} component-action`}
+						onClick={() => setEditorActive(!editorActive)}
+						type="button"
+					>
+						<Icon
+							spritemap={spritemap}
+							symbol={editorActive ? 'times' : 'drop'}
+						/>
+					</button>
+				</div>
+			)}
+
+			<div className="clay-color-swatch">
+				{colors.map((hex, i) => (
+					<div className="clay-color-swatch-item" key={i}>
+						<Splotch
+							active={i === activeSplotchIndex}
+							onClick={() => {
+								if (hex === 'FFFFFF') {
+									setEditorActive(true);
+								}
+
+								setActiveSplotchIndex(i);
+
+								setHue(tinycolor(hex).toHsv().h);
+
+								onChange(hex);
+							}}
+							value={hex}
+						/>
+					</div>
+				))}
+			</div>
+
+			{editorActive && (
+				<React.Fragment>
+					<div className="clay-color-map-group">
+						<GradientSelector
+							color={color}
+							hue={hue}
+							onChange={(saturation, visibility) => {
+								setNewColor(
+									tinycolor({
+										h: hue,
+										s: saturation,
+										v: visibility,
+									})
+								);
+							}}
+						/>
+
+						<div className="clay-color-map-values">
+							{rgbArr.map(([val, key]) => (
+								<RGBInput
+									key={key}
+									name={key}
+									onChange={newVal => {
+										const color = tinycolor({
+											b,
+											g,
+											r,
+											...newVal,
+										});
+
+										setHue(color.toHsv().h);
+										setNewColor(color);
+									}}
+									value={val}
+								/>
+							))}
+						</div>
+					</div>
+
+					<Hue
+						onChange={hue => {
+							setHue(hue);
+
+							setNewColor(tinycolor({h: hue, s, v}));
+						}}
+						value={hue}
+					/>
+
+					<div className="clay-color-footer">
+						<div className="form-group">
+							<div className="input-group">
+								<div className="input-group-item">
```

Comment:
When I am typing the first 3 digits it forces it to a color.

<img src="http://g.recordit.co/zPol7aIeiC.gif" />

Reply:
Yeah there seem to be a handful of interactive issues all stemming from `onChange`. I may need to set those to `onBlur` instead or I need to debounce setting the new color.